PR: https://github.com/odoo/odoo/pull/178200

From: f10bd8ddcf99c9247dca417fefbc260454d74ea1
From: Gorash
Date: 2024-10-15 09:33:02

Breaking data model changes scores: del:20 + add:0, change matches:
-    _inherit = 'account.move'
-    _inherit = 'account.move.line'
-    _inherit = "product.category"
-    _inherit = 'product.template'
-    _inherit = 'product.product'
-    _inherit = 'product.supplierinfo'
-    _inherit = 'purchase.order'
-    _inherit = 'purchase.order.line'
-    _inherit = 'res.company'
-    _inherit = 'res.config.settings'
-    _inherit = 'res.partner'
-    _inherit = 'stock.picking'
-    _inherit = 'stock.warehouse'
-    _inherit = "stock.return.picking"
-    _inherit = "stock.warehouse.orderpoint"
-    _inherit = 'stock.lot'
-    _inherit = 'procurement.group'
-    _inherit = 'stock.move'
-    _inherit = 'stock.rule'
-    _inherit = 'stock.valuation.layer'

Total Changes: 52

[IMP] *: adapt model class names to correspond to model names (apply script)

Part-of: odoo/odoo#178200
Related: odoo/enterprise#69762
Related: odoo/documentation#11115
Related: odoo/upgrade#6577
Related: odoo/design-themes#988
Related: odoo/upgrade-util#143

================================= pseudo patch: =================================

--- a/addons/purchase_stock/models/account_invoice.py
+++ b/addons/purchase_stock/models/account_invoice.py
@@ -7,7 +7,7 @@ from odoo.tools.misc import groupby
 
 
 class AccountMove(models.Model):
-    _inherit = 'account.move'
+    _inherit = ['account.move']
 
     def _stock_account_prepare_anglo_saxon_in_lines_vals(self):
         ''' Prepare values used to create the journal items (account.move.line) corresponding to the price difference

--- a/addons/purchase_stock/models/account_move_line.py
+++ b/addons/purchase_stock/models/account_move_line.py
@@ -8,7 +8,7 @@ from collections import defaultdict
 
 
 class AccountMoveLine(models.Model):
-    _inherit = 'account.move.line'
+    _inherit = ['account.move.line']
 
     def _get_valued_in_moves(self):
         self.ensure_one()

--- a/addons/purchase_stock/models/product.py
+++ b/addons/purchase_stock/models/product.py
@@ -6,7 +6,7 @@ from odoo.osv import expression
 
 
 class ProductCategory(models.Model):
-    _inherit = "product.category"
+    _inherit = ["product.category"]
 
     property_account_creditor_price_difference_categ = fields.Many2one(
         'account.account', string="Price Difference Account",
@@ -15,8 +15,7 @@ class ProductCategory(models.Model):
 
 
 class ProductTemplate(models.Model):
-    _name = 'product.template'
-    _inherit = 'product.template'
+    _inherit = ['product.template']
 
     property_account_creditor_price_difference = fields.Many2one(
         'account.account', string="Price Difference Account", company_dependent=True, ondelete='restrict',
@@ -34,8 +33,7 @@ class ProductTemplate(models.Model):
 
 
 class ProductProduct(models.Model):
-    _name = 'product.product'
-    _inherit = 'product.product'
+    _inherit = ['product.product']
 
     purchase_order_line_ids = fields.One2many('purchase.order.line', 'product_id', string="PO Lines") # used to compute quantities
 
@@ -91,8 +89,8 @@ class ProductProduct(models.Model):
         return expression.OR(domains) if domains else []
 
 
-class SupplierInfo(models.Model):
-    _inherit = 'product.supplierinfo'
+class ProductSupplierinfo(models.Model):
+    _inherit = ['product.supplierinfo']
 
     last_purchase_date = fields.Date('Last Purchase', compute='_compute_last_purchase_date')
     show_set_supplier_button = fields.Boolean(

--- a/addons/purchase_stock/models/purchase_order.py
+++ b/addons/purchase_stock/models/purchase_order.py
@@ -10,7 +10,7 @@ from odoo.tools.misc import OrderedSet
 
 
 class PurchaseOrder(models.Model):
-    _inherit = 'purchase.order'
+    _inherit = ['purchase.order']
 
     @api.model
     def _default_picking_type(self):

--- a/addons/purchase_stock/models/purchase_order_line.py
+++ b/addons/purchase_stock/models/purchase_order_line.py
@@ -6,7 +6,7 @@ from odoo.exceptions import UserError
 
 
 class PurchaseOrderLine(models.Model):
-    _inherit = 'purchase.order.line'
+    _inherit = ['purchase.order.line']
 
     def _ondelete_stock_moves(self):
         modified_fields = ['qty_received_manual', 'qty_received_method']

--- a/addons/purchase_stock/models/res_company.py
+++ b/addons/purchase_stock/models/res_company.py
@@ -5,7 +5,7 @@ from odoo import fields, models
 
 
 class ResCompany(models.Model):
-    _inherit = 'res.company'
+    _inherit = ['res.company']
 
     days_to_purchase = fields.Float(
         string='Days to Purchase',

--- a/addons/purchase_stock/models/res_config_settings.py
+++ b/addons/purchase_stock/models/res_config_settings.py
@@ -5,7 +5,7 @@ from odoo import api, fields, models
 
 
 class ResConfigSettings(models.TransientModel):
-    _inherit = 'res.config.settings'
+    _inherit = ['res.config.settings']
 
     module_stock_dropshipping = fields.Boolean("Dropshipping")
     days_to_purchase = fields.Float(

--- a/addons/purchase_stock/models/res_partner.py
+++ b/addons/purchase_stock/models/res_partner.py
@@ -8,7 +8,7 @@ from odoo import api, fields, models
 
 
 class ResPartner(models.Model):
-    _inherit = 'res.partner'
+    _inherit = ['res.partner']
 
     purchase_line_ids = fields.One2many('purchase.order.line', 'partner_id', string="Purchase Lines")
     on_time_rate = fields.Float(

--- a/addons/purchase_stock/models/stock.py
+++ b/addons/purchase_stock/models/stock.py
@@ -7,7 +7,7 @@ from odoo.osv.expression import AND
 
 
 class StockPicking(models.Model):
-    _inherit = 'stock.picking'
+    _inherit = ['stock.picking']
 
     purchase_id = fields.Many2one(
         'purchase.order', related='move_ids.purchase_line_id.order_id',
@@ -40,7 +40,7 @@ class StockPicking(models.Model):
 
 
 class StockWarehouse(models.Model):
-    _inherit = 'stock.warehouse'
+    _inherit = ['stock.warehouse']
 
     buy_to_resupply = fields.Boolean('Buy to Resupply', default=True,
                                      help="When products are bought, they can be delivered to this warehouse")
@@ -102,8 +102,8 @@ class StockWarehouse(models.Model):
         return res
 
 
-class ReturnPicking(models.TransientModel):
-    _inherit = "stock.return.picking"
+class StockReturnPicking(models.TransientModel):
+    _inherit = ["stock.return.picking"]
 
     def _prepare_move_default_values(self, return_line, new_picking):
         vals = super()._prepare_move_default_values(return_line, new_picking)
@@ -118,8 +118,8 @@ class ReturnPicking(models.TransientModel):
         return picking
 
 
-class Orderpoint(models.Model):
-    _inherit = "stock.warehouse.orderpoint"
+class StockWarehouseOrderpoint(models.Model):
+    _inherit = ["stock.warehouse.orderpoint"]
 
     show_supplier = fields.Boolean('Show supplier column', compute='_compute_show_suppplier')
     supplier_id = fields.Many2one(
@@ -245,7 +245,7 @@ class Orderpoint(models.Model):
 
 
 class StockLot(models.Model):
-    _inherit = 'stock.lot'
+    _inherit = ['stock.lot']
 
     purchase_order_ids = fields.Many2many('purchase.order', string="Purchase Orders", compute='_compute_purchase_order_ids', readonly=True, store=False)
     purchase_order_count = fields.Integer('Purchase order count', compute='_compute_purchase_order_ids')
@@ -270,7 +270,7 @@ class StockLot(models.Model):
 
 
 class ProcurementGroup(models.Model):
-    _inherit = 'procurement.group'
+    _inherit = ['procurement.group']
 
     purchase_line_ids = fields.One2many('purchase.order.line', 'group_id', string='Linked Purchase Order Lines', copy=False)
 

--- a/addons/purchase_stock/models/stock_move.py
+++ b/addons/purchase_stock/models/stock_move.py
@@ -8,7 +8,7 @@ from odoo.exceptions import UserError
 
 
 class StockMove(models.Model):
-    _inherit = 'stock.move'
+    _inherit = ['stock.move']
 
     purchase_line_id = fields.Many2one(
         'purchase.order.line', 'Purchase Order Line',

--- a/addons/purchase_stock/models/stock_replenish_mixin.py
+++ b/addons/purchase_stock/models/stock_replenish_mixin.py
@@ -3,8 +3,8 @@
 from odoo import api, fields, models
 
 
-class ProductReplenishMixin(models.AbstractModel):
-    _inherit = 'stock.replenish.mixin'
+class StockReplenishMixin(models.AbstractModel):
+    _inherit = ['stock.replenish.mixin']
 
     supplier_id = fields.Many2one('product.supplierinfo', string="Vendor")
     show_vendor = fields.Boolean(compute='_compute_show_vendor')

--- a/addons/purchase_stock/models/stock_rule.py
+++ b/addons/purchase_stock/models/stock_rule.py
@@ -12,7 +12,7 @@ from odoo.tools import groupby
 
 
 class StockRule(models.Model):
-    _inherit = 'stock.rule'
+    _inherit = ['stock.rule']
 
     action = fields.Selection(selection_add=[
         ('buy', 'Buy')

--- a/addons/purchase_stock/models/stock_valuation_layer.py
+++ b/addons/purchase_stock/models/stock_valuation_layer.py
@@ -2,7 +2,7 @@ from odoo import models
 
 
 class StockValuationLayer(models.Model):
-    _inherit = 'stock.valuation.layer'
+    _inherit = ['stock.valuation.layer']
 
     def _get_layer_price_unit(self):
         """ This function returns the value of product in a layer per unit, relative to the aml
